packages/nimble-components/src/table/specs/editable-table-ixd.md

```diff
@@ -0,0 +1,103 @@
+
+-   When data integrity is critical and the table should not be modified by users.
+-   When the table is used for displaying static or read-only data.
+-   When the user interface requires a more complex data entry form.
+
+## Anatomy
+
+![Editable table anatomy](./spec-images/editable-cell-anatomy.png)
+
+| Element               | Description                                                              |
+| --------------------- | ------------------------------------------------------------------------ |
+| Static cell value     | Non-editable display of cell value                                       |
+| Action button         | Button to open cell context menu. Appears on row hover or keyboard focus |
+| Editable cell control | Text or numeric control that supports entering or editing cell data      |
+
+## Behavior
+
+### States
+
+![Editable cell states](./spec-images/editable-cell-states.png)
+
+#### Error State
+
+![Editable cell error states](./spec-images/editable-cell-error-data.png)
+
+### ARIA Considerations
+
+-   Maintain compatibility with existing keyboard navigation behavior
+-   Consider resolving [existing ARIA gaps](https://github.com/ni/nimble/issues/2285)
+
+### Mouse Interactions
+
+![Editable cell mouse navigation](./spec-images/editable-cell-mouse-navigation.png)
+
+-   Editable table cells show the same row hover state as non-editable cells on mouse hover.
+    -   I.e. When hovering over a row, the action menu button appears. Depending on the table's selection mode, the row may be highlighted.
+-   Single clicking editable cell shows the cell focus state.
+-   Double clicking the editable cell transforms the cell into an input control in the focus state.
+-   Clicking away from a focused editable cell sets the value and transforms it into the Cell focus state.
+
+### Non-Mouse Interactions
+
+![Editable cell key navigation](./spec-images/editable-cell-key-navigation.png)
+
+-   When a cell has keyboard focus (Cell focus state), pressing `TAB` or `ENTER` transforms the cell into an input control in the focus state (Edit focus state).
+-   When a cell has Edit focus, pressing `TAB` moves the focus to the action button, or to the next available focus target.
+
+![Editable cell enter key](./spec-images/editable-cell-enter-key.png)
+
+-   When a cell has Edit focus, pressing `ENTER` sets the value and transforms it into the Cell focus state.
+
+![Editable cell escape key](./spec-images/editable-cell-escape-key.png)
+
+-   When a cell has Edit focus, pressing `ESCAPE` reverts any value change and transforms it into the Cell focus state.
+
+![Editable cell data error](./spec-images/editable-cell-error-data.png)
+
+-   When a cell has Edit focus, pressing `ENTER` validates the value, and if invalid, transforms the cell into the Cell error state.
+
+### Out of scope
+
+The client application is responsible for defining and implementing workflows for adding, deleting, and moving rows, as well as saving data.
```

Comment:
We probably need to think through situations like if editing a value causes its sort order or grouping to change. Do we have a recommended experience? Are we ok with the value you just edited disappearing off screen? Should we scroll to it? Wait to apply changes until they finish a batch of edits?

Reply:
The least disruptive option is keep the new edited value in the same relative location after completing the edit, even if the column sort would move the row.

If that's not possible, and the row needs to immediately sort to the correct position, some affordance to indicate that the row is moving would be helpful. E.g., an animation of the row moving to it's new location. We shouldn't scroll to the new position.